drift-protocol/src/main/java/com/facebook/drift/protocol/TBinaryProtocol.java

```diff
@@ -391,6 +393,16 @@ public ByteBuffer readBinary()
         return ByteBuffer.wrap(buf);
     }
 
+    @Override
+    public int readBinary(byte[] buf, int offset)
+            throws TException
+    {
+        int size = checkSize(readI32());
+        checkArgument((buf.length - offset) >= size, "Buffer should have enough space to read all data from wire");
```

Comment:
When checking boundaries it is generally very useful to include concrete numbers in the message. It helps a lot when debugging. How about changing this message to:

`"Binary is too large to be read into buffer: binary size: %s, buffer size: %s, buffer offset: %s", size, buf.length, offset`